File: src/main/java/de/twometer/amongus/gui/FixSabotagePage.java
package de.twometer.amongus.gui;

import de.twometer.amongus.core.AmongUs;
import de.twometer.amongus.event.SabotageEvent;
import de.twometer.amongus.model.Sabotage;
import de.twometer.amongus.net.NetMessage;
import org.greenrobot.eventbus.Subscribe;

public class FixSabotagePage extends BasePage {

    private final Sabotage sabotage;

    public FixSabotagePage(Sabotage sabotage) {
        super("Tasks/Fix" + sabotage.name() + ".html");
        this.sabotage = sabotage;
    }

    @Override
    public void onDomReady() {
        if (AmongUs.get().getSession().currentSabotage == Sabotage.O2) {
            context.call("setCode", AmongUs.get().getSession().currentSabotageCode);
        }
    }

    public void setFixing(boolean fixing) {
        AmongUs.get().getClient().sendMessage(new NetMessage.FixSabotage(AmongUs.get().getSession().currentSabotage, fixing));
    }

    @Subscribe
    public void onSabotageFixed(SabotageEvent event) {
        if (amongUs.getSession().currentSabotage == null) {
            AmongUs.get().getScheduler().runLater(800, () -> {
                if (AmongUs.get().getStateController().isRunning())
                    this.goBack();
            });
        }
    }

    @Override
    protected boolean escapeGoesBack() {
        return true;
    }

}

File: src/main/kotlin/de/twometer/amogus/gui/FixSabotagePage.kt
package de.twometer.amogus.gui

import de.twometer.amogus.client.AmongUsClient
import de.twometer.amogus.client.SabotageGameObject
import de.twometer.amogus.model.SabotageType
import de.twometer.amogus.net.SabotageFix
import de.twometer.neko.events.Events
import mu.KotlinLogging

private val logger = KotlinLogging.logger {}

class FixSabotagePage(private val sabotage: SabotageGameObject) : BasePage("Tasks/Fix${sabotage.sabotageType}.html") {

    override fun onLoaded() {
        if (AmongUsClient.currentSabotage == SabotageType.O2)
            call("setCode", AmongUsClient.currentSabotageCode)
        Events.register(this)
    }

    override fun onUnloaded() {
        Events.unregister(this)
    }

    fun setFixing(fix: Boolean) {
        AmongUsClient.send(SabotageFix(sabotage.location, fix))
        logger.debug { "Fix state $fix at ${sabotage.location}" }
    }

}